packages/yarnpkg-core/sources/tgzUtils.ts

```diff
@@ -58,71 +61,79 @@ export async function extractArchiveTo<T extends FakeFS<PortablePath>>(tgz: Buff
     return false;
   }
 
-  parser.on(`entry`, (entry: any) => {
-    if (ignore(entry)) {
-      entry.resume();
+  parser.on(`entry`, (header, stream, next) => {
+    if (ignore(header)) {
+      next();
       return;
     }
 
-    const parts = ppath.normalize(npath.toPortablePath(entry.path)).replace(/\/$/, ``).split(/\//g);
+    const parts = ppath.normalize(npath.toPortablePath(header.name)).replace(/\/$/, ``).split(/\//g);
     if (parts.length <= stripComponents) {
-      entry.resume();
+      stream.resume();
+      next();
       return;
     }
 
     const slicePath = parts.slice(stripComponents).join(`/`) as PortablePath;
     const mappedPath = ppath.join(prefixPath, slicePath);
 
-    const chunks: Array<Buffer> = [];
-
     let mode = 0o644;
 
     // If a single executable bit is set, normalize so that all are
-    if (entry.type === `Directory` || (entry.mode & 0o111) !== 0)
+    if (header.type === `directory` || ((header.mode ?? 0) & 0o111) !== 0)
       mode |= 0o111;
 
-    entry.on(`data`, (chunk: Buffer) => {
-      chunks.push(chunk);
-    });
+    switch (header.type) {
+      case `directory`: {
+        targetFs.mkdirpSync(ppath.dirname(mappedPath), {chmod: 0o755, utimes: [defaultTime, defaultTime]});
 
-    entry.on(`end`, () => {
-      switch (entry.type) {
-        case `Directory`: {
-          targetFs.mkdirpSync(ppath.dirname(mappedPath), {chmod: 0o755, utimes: [defaultTime, defaultTime]});
+        targetFs.mkdirSync(mappedPath);
+        targetFs.chmodSync(mappedPath, mode);
+        targetFs.utimesSync(mappedPath, defaultTime, defaultTime);
+        next();
+      } break;
 
-          targetFs.mkdirSync(mappedPath);
-          targetFs.chmodSync(mappedPath, mode);
-          targetFs.utimesSync(mappedPath, defaultTime, defaultTime);
-        } break;
+      case `file`: {
+        targetFs.mkdirpSync(ppath.dirname(mappedPath), {chmod: 0o755, utimes: [defaultTime, defaultTime]});
 
-        case `OldFile`:
-        case `File`: {
-          targetFs.mkdirpSync(ppath.dirname(mappedPath), {chmod: 0o755, utimes: [defaultTime, defaultTime]});
+        const chunks: Array<Buffer> = [];
 
+        stream.on(`data`, (chunk: Buffer) => chunks.push(chunk));
+        stream.on(`end`, () => {
+          targetFs.createWriteStream(mappedPath);
```

Comment:
```suggestion
```
Assuming this was an accident

Reply:
Oh yeah, that's a leftover that indeed shouldn't be there.